packages/vaex-core/vaex/dataframe.py

```diff
@@ -3954,8 +3956,14 @@ def sort(self, by, ascending=True, kind='quicksort'):
         :param str kind: kind of algorithm to use (passed to numpy.argsort)
         '''
         self = self.trim()
-        values = self.evaluate(by)
-        indices = np.argsort(values, kind=kind)
+        if not isinstance(by, list):
+            values = self.evaluate(by)
+            indices = np.argsort(values, kind=kind)
+        if isinstance(by, list):
+            by = _ensure_strings_from_expressions(by)[::-1]
+            by = ', '.join(by)
```

Comment:
This is actually not supported, I'm working on a parallel evalulate that takes a list of expressions. I'm ok with merging this, but we need to put a reminder
```suggestion
            by = ', '.join(by)  # TODO: this is using an unsupported features, when evaluate takes a list of expressions, we should change this.
```

Reply:
Note that the unittest will catch it, but it's good to have this in the code.

Reply:
Fair point - comment added.